CondCore/PopCon/src/OnlinePopCon.cc

```diff
@@ -0,0 +1,116 @@
+      : m_targetSession(),
+        m_targetConnectionString(pset.getUntrackedParameter<std::string>("targetDBConnectionString", "")),
+        m_authPath(pset.getUntrackedParameter<std::string>("authenticationPath", "")),
+        m_authSys(pset.getUntrackedParameter<int>("authenticationSystem", 1)),
+        m_recordName(pset.getParameter<std::string>("record")),
+        m_useLockRecors(pset.getUntrackedParameter<bool>("useLockRecords", false)) {
+    edm::LogInfo("OnlinePopCon")
+        << "This is OnlinePopCon (Populator of Condition) v" << s_version << ".\n"
+        << "Please report any problem and feature request through the JIRA project CMSCONDDB.\n";
+  }
+
+  OnlinePopCon::~OnlinePopCon() {
+    if (!m_targetConnectionString.empty()) {
+      m_targetSession.transaction().commit();
+    }
+  }
+
+  cond::persistency::Session OnlinePopCon::preparePopCon() {
+    // Initialization almost identical to PopCon
+    const std::string& connectionStr = m_dbService->session().connectionString();
+    m_dbService->forceInit();  // FIXME: should we run this?
+    std::string tagName = m_dbService->tag(m_recordName);
+    m_tagInfo.name = tagName;
+    if (m_targetConnectionString.empty()) {
+      m_targetSession = m_dbService->session();
+      m_dbService->startTransaction();
+    } else {
+      cond::persistency::ConnectionPool connPool;
+      connPool.setAuthenticationPath(m_authPath);
+      connPool.setAuthenticationSystem(m_authSys);
+      connPool.configure();
+      m_targetSession = connPool.createSession(m_targetConnectionString);
+      m_targetSession.transaction().start();
+    }
+    if (m_targetSession.existsDatabase() && m_targetSession.existsIov(tagName)) {
+      cond::persistency::IOVProxy iov = m_targetSession.readIov(tagName);
+      m_tagInfo.size = iov.sequenceSize();
+      if (m_tagInfo.size > 0) {
+        m_tagInfo.lastInterval = iov.getLast();
+      }
+      m_dbService->logger().logInfo() << "OnlinePopCon::preparePopCon";
+      m_dbService->logger().logInfo() << "  destination DB: " << connectionStr;
+      m_dbService->logger().logInfo() << "  target DB: "
+                                      << (m_targetConnectionString.empty() ? connectionStr : m_targetConnectionString);
+      m_dbService->logger().logInfo() << "  TAG: " << tagName << ", last since/till: " << m_tagInfo.lastInterval.since
+                                      << "/" << m_tagInfo.lastInterval.till;
+      m_dbService->logger().logInfo() << "  size: " << m_tagInfo.size;
+    } else {
+      m_dbService->logger().logInfo() << "OnlinePopCon::preparePopCon";
+      m_dbService->logger().logInfo() << "  destination DB: " << connectionStr;
+      m_dbService->logger().logInfo() << "  target DB: "
+                                      << (m_targetConnectionString.empty() ? connectionStr : m_targetConnectionString);
+      m_dbService->logger().logInfo() << "  TAG: " << tagName << "; First writer to this new tag.";
+    }
```

Comment:
I would put the common logging for these 2 cases before the if so it isn't duplicated